Hometask_2/src/index.ts

```diff
@@ -0,0 +1,200 @@
+console.log("Start hometrask");
+
+function excercise4() {
+  let firstVariable: [number, number];
+  let secondVariable: [number, number];
+
+  firstVariable = [1, 1];
+  secondVariable = [4, 5];
+
+  function calculateDistance(
+    firstVariable: [number, number],
+    secondVariable: [number, number]
+  ): number {
+    const x1 = firstVariable[0];
+    const y1 = secondVariable[0];
+    const x2 = firstVariable[1];
+    const y2 = secondVariable[1];
+    const dx = x2 - x1;
+    const dy = y2 - y1;
+    return Math.sqrt(dx * dx + dy * dy);
+  }
+
+  const result = calculateDistance(firstVariable, secondVariable);
+  console.log(`Distance between poionts____${result}`);
+}
+excercise4();
+
+function excercise5() {
+  type TPoint = {
+    x: number;
+    y: number;
+  };
+
+  const p1: TPoint = { x: 1, y: 1 };
+  const p2: TPoint = { x: 4, y: 5 };
+
+  function distance(p1: TPoint, p2: TPoint): number {
+    const { x: x1, y: y1 } = p1;
+    const { x: x2, y: y2 } = p2;
+    const deltaX = x2 - x1;
+    const deltaY = y2 - y1;
+    const distance = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
+    return distance;
+  }
+
+  const result = distance(p1, p2);
+  console.log(`Distance between poionts____${result}`);
+}
+excercise5();
+
+function excercise6() {
+  const PI = 3.14;
+
+  function getCircleAria(radius: number): number {
+    return PI * (radius * radius);
+  }
+  const radius = 5;
+  const circleArea = getCircleAria(radius);
+
+  console.log(`The area of the circke with radius ${radius} = ${circleArea}`);
+
+  console.log(`The type of PI is ${typeof PI}`);
+
+  type TPerson = {
+    name: string;
+    age: number;
+  };
+
+  const person: TPerson = {
+    name: "Alex",
+    age: 40,
+  };
+
+  function incrementsAge(person: TPerson): void {
+    person.age = person.age + 1;
+  }
+  incrementsAge(person);
+  console.log(`Name: ${person.name}, Age: ${person.age}`);
+}
+excercise6();
+
+function excercise7() {
+  function map(arr: number[], fn: (num: number) => number): number[] {
```

Comment:
Some other way to implement this is to create type annotation for this function and use it here and in line 94